sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -455,6 +465,16 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
     this
   }
 
+  private[sql] def foreachConnect(writer: PythonForeachWriter): DataStreamWriter[T] = {
```

Comment:
Will this work for Scala connect too? Otherwise earlier name `foreachPython` sounds better.

Reply:
Yes actually this change is due to the change I added in scala foreach: https://github.com/apache/spark/pull/41129/files#r1194152295

The writer will be a generic ForeachWriter[Any]. Here I put PythonForeachWriter because at this stage it doesn't need to be generic